collector/monitors/slashing_monitor.go

```diff
@@ -25,42 +26,58 @@ const (
 )
 
 type SlashingMonitor struct {
-	metrics              map[MetricName]float64
-	metricVectors        map[MetricName]MetricVector
+	metrics       map[MetricName]MetricValue
+	metricVectors map[MetricName]*MetricVector
+	// tmp* for 2stage nonblocking update data
```

Comment:
I don't think we need these values at the struct level? They can easily be declared as local variables inside `Handler()`.